docs/workshop-hello-world.md

```diff
@@ -0,0 +1,269 @@
+If you've already completed the steps above, you can skip to the next section.
+
+- Create a new branch in git to push your changes to
+- Open up `deployments/stacks/dpe-k8s-deployments/main.tf`
+  - You'll notice a bunch of modules. These are the resources we are already deploying to the kubernetes cluster, wrapped in something called a "module"
+  - For this tutorial we will create the resource files directly here for simplicity
+
+- Create 2 sections in the file:
+
+1) A namespace: <https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/>
+
+Update `my-cool-namespace-resource` and `my-cool-namespace` to whatever values you'd like. In this
+case `my-cool-namespace-resource` is the identifier in which terraform is referencing this specific
+resource. `my-cool-namespace` on the otherhand is the name of the namespace you'll be
+creating within the kubernetes cluster.
+
+```terraform
+resource "kubernetes_namespace" "my-cool-namespace-resource" {
+  metadata {
+    name = "my-cool-namespace"
+  }
+}
+```
+
+2) A Deployment: <https://kubernetes.io/docs/concepts/workloads/controllers/deployment/>
+
+You'll want to update the reference to your terraform namespace resource here:
+
+- If you updated the terraform identifier `my-cool-namespace-resource` to something else, make sure they match
+```terraform
+resource "kubectl_manifest" "my-deployment" {
+  depends_on = [kubernetes_namespace.my-cool-namespace-resource]
+
+  yaml_body = <<YAML
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  namespace: ${kubernetes_namespace.my-cool-namespace-resource.metadata.0.name}
+  name: flask-helloworld
+  labels:
+    app: flask-helloworld
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: flask-helloworld
+  template:
+    metadata:
+      labels:
+        app: flask-helloworld
+    spec:
+      containers:
+      - name: flask
+        image: digitalocean/flask-helloworld:latest
+        ports:
+        - containerPort: 5000
+YAML
+}
+```
+
+- Commit the changes and push the changes to github
+- Create a pull request to main
+- In the git checks that run you'll see a section that says something like: `spacelift/dpe-dev-kubernetes-deployments — 2 to add, 6 to change, 0 to destroy`
+
+![Github check](./workshop-resources/github-check.png)
+
+- Click on `Details` and on the next screen `Deploy`
+- A job will be kicked off and run within Spacelift to deploy the resources. This takes a few minutes.
+
+### Verifying your deployed resources on the kubernetes cluster
+
+- Inside of `k9s` check that your namespace exists `:ns`
+- Check that your deployment is there `:deployments`
+- Finally let's port-forward to the pod that was created `:pods`
+- Navigate to the pod in your namespace and use `shift + f` to start a port forward session
+- Open a web browser and go to `localhost:5000` and verify you may see `Hello, World!`
+
+
+That's it! You've deployed a resource to our kubernetes cluster!
```

Comment:
It might be helpful, to also have some instructions on how to tear this all down. I'd like to stop forwarding this application to 5000 and also remove the namespace. Would that just involve deleting the branch?

Reply:
The resources will get cleaned up and removed the next time that the terraform pipeline runs with a change. I'm not too concerned with these resources lingering around until then.

There are some concerns though around stepping on the toes of others since terraform is expecting a single source of truth for it's state. I'll need to think through some more long term deployment strategies here. Like creating specific spacelift stacks during testing so we can deploy independently from other developers.

I'll add comments though